lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/UpdateTaskHost.cs

```diff
@@ -104,19 +113,53 @@ public byte[] Call(byte[] memento)
                 }
             }
 
-            MapInputWithControlMessage<TMapInput> stopMessage =
+            if (!_break)
+            {
+                MapInputWithControlMessage<TMapInput> stopMessage =
                     new MapInputWithControlMessage<TMapInput>(MapControlMessage.Stop);
-            _dataAndControlMessageSender.Send(stopMessage);
+                _dataAndControlMessageSender.Send(stopMessage);
+            }
 
             _resultHandler.Dispose();
+            _stopped = true;
             return null;
         }
 
         /// <summary>
+        /// Received close event. If it is closed command sent from driver, set _break to true to stop the iteration in Call method
+        /// After _enforceCloseTimeout, if the task is still not closed, throw exception to enforce stop. 
+        /// </summary>
+        /// <param name="closeEvent"></param>
+        public void OnNext(ICloseEvent closeEvent)
+        {
+            if (closeEvent.Value.IsPresent() && Encoding.UTF8.GetString(closeEvent.Value.Value).Equals(TaskManager.CloseTaskByDriver))
+            {
+                Logger.Log(Level.Info, "The task received close event with message: {0}.", Encoding.UTF8.GetString(closeEvent.Value.Value));
+                _break = true;
+            }
+            Thread.Sleep(_enforceCloseTimeout);
```

Comment:
Here is the scenario: driver sends close event to task for it to e closed, however, the task is hung in Call() method for a blocking call. There is no way to signal the close handler to take some action. In this case, we would just want to enforce the task to close by throwing an exception. In this case, the accuracy of the time is not very critical as we just want to eventually close the task after waiting for some time if task is not returned from Call() yet. 

Please describe what's the potential problem here and any better idea.